stats/stats_test.go

```diff
@@ -720,7 +720,64 @@ func TestNaN(t *testing.T) {
 	}
 }
 
+func TestBucketLookUp(t *testing.T) {
+	var tests = []struct {
+		input float64 // input
+		start float64 // start
+		end   float64 // end
+
+	}{
+		{input: 11, start: 10, end: 11},
+		{input: 171, start: 160, end: 180},
+		{input: 180, start: 160, end: 180},
+		{input: 801, start: 800, end: 900},
+		{input: 900, start: 800, end: 900},
+		{input: 999, start: 900, end: 1000},
+		{input: 999, start: 900, end: 1000},
+		{input: 999.99, start: 900, end: 1000},
+		{input: 1000, start: 900, end: 1000},
+		{input: 1000.01, start: 1000, end: 2000},
+		{input: 1001, start: 1000, end: 2000},
+		{input: 1001.1, start: 1000, end: 2000},
+		{input: 1999.99, start: 1000, end: 2000},
+		{input: 2000, start: 1000, end: 2000},
+		{input: 2001, start: 2000, end: 3000},
+		{input: 2001.1, start: 2000, end: 3000},
+		{input: 100000.0001, start: 75000, end: 100000},
```

Comment:
using the exact number that gets subtracted is a bit brittle - stick to X is before and X.01 is above
ie change this to test
99999.99
100000
100000.01
(and if you want
100000.99
100001
)

Reply:
Should I do that for all boundary values?